packit/utils/specfile_helper.py

```diff
@@ -0,0 +1,115 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+from typing import List, Optional
+
+from specfile import Specfile
+from specfile.exceptions import SpecfileException
+
+from packit.patches import PatchMetadata
+from packit.exceptions import PackitException
+
+try:
+    from rebasehelper.plugins.plugin_manager import plugin_manager
+except ImportError:
+    from rebasehelper.versioneer import versioneers_runner
+
+
+logger = logging.getLogger(__name__)
+
+
+class SpecfileHelper:
```

Comment:
Is having this class really required? I see there are only class- and static methods. Why not use functions instead?

Reply:
No, it's not, I'd just have to come up with meaningful function names :slightly_smiling_face:

Reply:
You could rely on the module to namespace these. IMHO, if function names make sense in the context of the module that's just fine.

```python
from packit.utils import specfile_helper

specfile_helper.set_patches()
```

Reply:
Is it possible to flexmock a function though? It doesn't seem to work no matter what I try :frowning_face:

Reply:
Yes, it is, you have to flexmock it on the module (there may be other ways, but I haven't found anything else that works). An example from my recent memory, that mocks a function `group` from module `celery`:

https://github.com/packit/packit-service/blob/main/tests/unit/test_jobs.py#L2444

Reply:
Thanks, I tried that, but it doesn't work.

```python
from packit.utils import specfile_helper
...
flexmock(specfile_helper, get_upstream_version=lambda **kw: None)
```

I tried other variations as well (mocking `packit` directly, using `should_receive()` etc.). No idea what's wrong.

Reply:
Can you try `flexmock(packit.utils.specfile_helper).should_receive("get_upstream_version").and_return(None)`? That would match the link I sent. Flexmock is sometimes weird so these two ways may actually be different, who knows

Reply:
I tried that already, it's still the same, it calls the original function which causes the test to fail.

Reply:
And of course everything works when I switch back to static method :man_shrugging:

Reply:
Thanks, that makes sense, and it works :slightly_smiling_face: